src/main/java/org/wso2/carbon/securevault/hashicorp/repository/HashiCorpSecretRepository.java

```diff
@@ -179,4 +192,130 @@ public SecretRepository getParent() {
+
+        if(new File(carbonHome + File.separator + textFilePersist).exists()){
+            persistToken = true;
+        }
+
+        tokenFile = new File(carbonHome + File.separator + textFileName);
+        if(tokenFile.exists()){
+            rootToken = readToken(tokenFile);
+
+            if(!persistToken) {
+                if(!renameConfigFile(textFileName_tmp)){
+                    throw new HashiCorpVaultException("Error in renaming password config file");
+                }
+            }
+        } else {
+            tokenFile = new File(carbonHome + File.separator + textFileName_tmp);
+            if(tokenFile.exists()){
+                rootToken = readToken(tokenFile);
+
+                if(!persistToken) {
+                    if(deleteConfigFile()){
+                        throw new HashiCorpVaultException("Error in deleting password config file");
+                    }
+                }
+            }else {
+                tokenFile = new File(carbonHome + File.separator + textFilePersist);
+                if(tokenFile.exists()){
+                    rootToken = readToken(tokenFile);
+
+                    if(!persistToken) {
+                        if(deleteConfigFile()){
+                            throw new HashiCorpVaultException("Error in deleting password config file");
+                        }
+                    }
+                }else {
+                    Console console;
+                    char[] token;
+                    if ((console = System.console()) != null && (token = console.readPassword("[%s]",
+                            "Enter the Root Token: ")) != null) {
+                        rootToken = String.valueOf(token);
+                    }
+                }
+            }
+        }
+    }
+
+    /**
+     * Set the name for the text file which contains the root token.
+     * For Linux: The file name should be hashicorpRootToken-tmp/ hashicorpRootToken-persist.
+     * For Windows: The file name should be hashicorpRootToken-tmp.txt/ hashicorpRootToken-persist.txt.
+     *
+     */
+    private void setTextFileName() {
+        String osName = System.getProperty("os.name");
+        if(osName.toLowerCase().contains("win")){
+            textFileName = "hashicorpRootToken.txt";
+            textFileName_tmp = "hashicorpRootToken-tmp.txt";
+            textFilePersist = "hashicorpRootToken-persist.txt";
+        }else {
+            textFileName = "hashicorpRootToken";
+            textFileName_tmp = "hashicorpRootToken-tmp";
+            textFilePersist = "hashicorpRootToken-persist";
+        }
+    }
+
+    /**
+     * Util method to Read the root token from the text file.
+     *
+     * @param tokenFile File containing the root token.
+     * @return The read token.
+     * @throws HashiCorpVaultException when an error occurred while reading the root token.
+     */
+    private String readToken(File tokenFile) throws HashiCorpVaultException{
+        String tokenReadFromFile;
```

Comment:
Newline after method name.
Update in similar places

Reply:
Done